daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -345,6 +347,63 @@ allocateParty args name = do
         setRequestBodyJSON $ AllocatePartyRequest {identifierHint = text, displayName = text}
   return party
 
+runLedgerReset :: LedgerFlags -> IO ()
+runLedgerReset flags = do
+  putStrLn "Resetting ledger."
+  args <- getDefaultArgs flags
+  reset args
+
+reset :: LedgerArgs -> IO ()
+reset args =
+  case api args of
+    Grpc ->
+      runWithLedgerArgs args $ do
+        parties <- map L.party <$> L.listKnownParties
+        unless (null parties) $ do
+          ledgerId <- L.getLedgerIdentity
+          L.setToken
+            (L.Token $
+             T.unpack $
+             tokenFor
+               [TL.toStrict $ L.unParty p | p <- parties]
+               (TL.toStrict $ L.unLedgerId ledgerId)
+               "ledger-reset") $ do
+            activeContracts <-
+              L.getActiveContracts
+                ledgerId
+                (TransactionFilter $
+                 Map.fromList [(L.unParty p, Just $ Filters Nothing) | p <- parties])
+                (L.Verbosity False)
+            let cmds =
+                  L.Commands
+                    { coms =
+                        [ L.ExerciseCommand
```

Comment:
Have you checked how well this works with large ACS sizes? I know that in the past, sending a large number of commands in one transaction caused issues and you were better off batching them in a few transactions but it’s been a while so maybe it’s fine now.

Reply:
I didn't test. Who could know about the current state?

Reply:
Maybe @remyhaemmerle-da I think at least the obvious stackoverflows are fixed now. Not sure what other issues are lurking there.

Reply:
Only the very obvious stackoverflows are fixed, but the compiler still can blow up if the expression it generates are too big. 
I am not quite sure about what you trying to achieve here, but I think that in general one should try to make the smallest possible transactions. In other words one should use several commands in a submission only if one needs a atomic commit of those commands.

Reply:
one transaction per command is super slow here. I would batch it in batches of 100 or so

Reply:
Yes it is probably quite slow indeed.

Reply:
ok, we batch them now in chunks of 100.